### DISCUSSION TITLE
Работа zapret с оператором мтс

### DISCUSSION THREAD

**Discussion by electrifying** [bol-van/zapret#229] [Q&A]
Zapret развернут на роутере keenetic ultra(KN-1811) до вчерашнего дня заблокированные сайты и ютуб работал нормально, сегодня уже нет использовал такие флаги
`--dpi-desync=disorder --dpi-desync-ttl=2 --dpi-desync-split-pos=1`
Как понять, это провайдер что-то у себя поменял и теперь флаги неактуальные или это просто кинетик почему-то перестал стартовать zapret? Как можно это узнать? 

В логах вижу такую ошибку

```
Opkg::Manager: /opt/etc/ndm/netfilter.d/000-zapret.sh: /opt/etc/ndm/netfilter.d/000-zapret.sh: 
exit: line 2: Illegal number: 0.

Opkg::Manager: /opt/etc/ndm/netfilter.d/000-zapret.sh: exit code 2.
```

есть варианты как можно это исправить? 

В 000-zapret.sh прописан вот такой скрипт

```
#!/bin/sh
[ "$type" == "ip6tables" ] && exit 0
[ "$table" != "mangle" ] && exit 0
/opt/zapret/init.d/sysv/zapret restart-fw

```
ValdikSS рекомендует использовать для провайдера мтс/мгтс флаги -5 -e1 какие эквивалентные флаги будут для zapret?

**Comment by numbereleven-a** [comment]


**Reply by electrifying** [reply]
> --dpi-desync-fooling=md5sig

Блокчек же предупреждает, что этот флаг может сломать работу сайтов. Попробуйте его убрать.Еще стоит проверить сайт rutor.info у него нет https, поэтому если флаги настроенные неправильно он не откроется, либо откроется но после пары рефрешей попадет в заглушку провайдера

**Reply by bol-van** [reply]
вы неверно поняли. этот флаг не ломает сайты
но его может быть недостаточно
если сервер не на linux, ос по барабану этот мд5
фейк будет принят и сломает соединение
поэтому его лучше сочетать с ттл. что-то да сработает
но ттл нельзя копипастить

**Comment by bol-van** [comment]
второй вариант делает не меньше, чем гудбай
проблема может быть в ттл, который вы скопипастили
или в хостлисте
стоит его временно убрать и проверить
ну и конечно убедиться,  что ипв6 не зря отключен

**Reply by bol-van** [reply]
аналог -e 1 -f 1  это
nfqws --dpi-desync=split2 --dpi-desync-split-pos=1
fake   тут лишний, а без него не нужны ограничители

если не работает берите tcpdump и смотрите фрагментацию на внешнем интерфейсе роутера
сравните ее с фрагментацией  гудбая
смотрите так же лог nfqws --debug

тут народ пытается прикрутиться к разныи стокам по разным инструкциям
если это ваш случай., то тем более надо смотреть работает оно как задумано или нет

**Reply by electrifying** [reply]
> fake тут лишний

На самом деле если провайдер мтс эти два флага не оказывают никакого влияния и заблокированные сайты не заработают, а вот с фейком уже другое дело. Даже если 2 раза прогнать блокчек где в каждом будет по 8 прогонов, то ни в одном из них не выпадет для мтс стратегия без фейка

**Reply by bol-van** [reply]
я лишь перевожу опции гудбая
с  -e 1 -f 1
он не шлет фейков. насколько понимаю гудбай

**Reply by wowa-2017** [reply]
А кто-то знает, гудбай - это аналог twps или nfqws или гибрид? Вот бы кто-нибудь заметочку где-нить написал сравнительную...

**Reply by bol-van** [reply]
гудбай работает на пакетном уровне, как и nfqws

**Comment by numbereleven-a** [comment]


**Reply by electrifying** [reply]
> да и в целом больше ничего кроме рутрекера не работает)

ну когда ttl=0 ничего удивительного нет

**Comment by iliakuzmich** [comment]
Всем привет
Вдруг кому-то поможет, оператор мтс/мгтс , по состоянию на 14.09.2024 у МЕНЯ заработали параметры через install_easy приведенные ниже, исходи из особенностей пути трафика и индивидуальной ситуации - не гарантируется работоспособность у Вас

NFQWS_OPT_DESYNC="--dpi-desync=fake,split2 --dpi-desync-ttl=1 --dpi-desync-split-pos=1"
NFQWS_OPT_DESYNC_HTTP=" --dpi-desync=split2 --dpi-desync-split-seqovl=9 --dpi-desync-split-pos=10 --wssize 1:6"
NFQWS_OPT_DESYNC_HTTPS="--dpi-desync=split2 --dpi-desync-split-seqovl=9 --dpi-desync-split-pos=10 --wssize 1:6"
NFQWS_OPT_DESYNC_HTTP6=" --dpi-desync=split2 --dpi-desync-split-seqovl=9 --dpi-desync-split-pos=10 --wssize 1:6"
NFQWS_OPT_DESYNC_HTTPS6=" --dpi-desync=split2 --dpi-desync-split-seqovl=9 --dpi-desync-split-pos=10 --wssize 1:6"
NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake --dpi-desync-repeats=6"
NFQWS_OPT_DESYNC_QUIC6=""

**Reply by iliakuzmich** [reply]
Спасибо
Правильно понимаю, что если NFQWS_OPT_DESYNC_HTTP6 не указан и не указан NFQWS_OPT_DESYNC_HTTP - то берется значение NFQWS_OPT_DESYNC ?

**Reply by bol-van** [reply]
Верно. Вот полный код присваивания умолчаний :
```
	NFQWS_OPT_DESYNC_HTTP="${NFQWS_OPT_DESYNC_HTTP:-$NFQWS_OPT_DESYNC}"
	NFQWS_OPT_DESYNC_HTTPS="${NFQWS_OPT_DESYNC_HTTPS:-$NFQWS_OPT_DESYNC}"
	NFQWS_OPT_DESYNC_HTTP6="${NFQWS_OPT_DESYNC_HTTP6:-$NFQWS_OPT_DESYNC_HTTP}"
	NFQWS_OPT_DESYNC_HTTPS6="${NFQWS_OPT_DESYNC_HTTPS6:-$NFQWS_OPT_DESYNC_HTTPS}"
	NFQWS_OPT_DESYNC_QUIC6="${NFQWS_OPT_DESYNC_QUIC6:-$NFQWS_OPT_DESYNC_QUIC}"

```

**Reply by iliakuzmich** [reply]
Понял, благодарю, так стало понятнее, что мой конфиг избыточен
